core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java

```diff
@@ -102,16 +106,43 @@ public static Pathfinding.Result<SignalingRoute> runPathfinding(
             remainingDistanceEstimator = new RemainingDistanceEstimator(waypoints.get(1));
         }
 
+        Pathfinding.Result<SignalingRoute> pathfinding = computePaths(infra, waypoints, loadingGaugeConstraints,
+                electrificationConstraints, remainingDistanceEstimator);
         // Compute the paths from the entry waypoint to the exit waypoint
-        return new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
+        return pathfinding;
+    }
+
+    private static Pathfinding.Result<SignalingRoute> computePaths(SignalingInfra infra,
+             ArrayList<Collection<Pathfinding.EdgeLocation<SignalingRoute>>> waypoints,
+             LoadingGaugeConstraints loadingGaugeConstraints,
+             ElectrificationConstraints electrificationConstraints,
+             RemainingDistanceEstimator remainingDistanceEstimator) {
+
+        var path = new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
                 .setEdgeToLength(route -> route.getInfraRoute().getLength())
-                .addBlockedRangeOnEdges(loadingGaugeConstraints)
+                .setRemainingDistanceEstimator(remainingDistanceEstimator);
+        var pathfinding = path.addBlockedRangeOnEdges(loadingGaugeConstraints)
                 .addBlockedRangeOnEdges(electrificationConstraints)
-                .setRemainingDistanceEstimator(remainingDistanceEstimator)
                 .runPathfinding(waypoints);
+        // handling errors
+        if (pathfinding == null) {
+            var pathError = path.addBlockedRangeOnEdges(loadingGaugeConstraints).runPathfinding(waypoints);
+            if (pathError != null) {
+                throw new NoPathFoundError("No path could be found due to Gauge");
+            } else {
```

Comment:
Nitpicking, but nested `if` makes this harder to read. It's best to return or throw early, that way we don't need `else`